### PR TITLE
Add parse structure for testing

### PR DESCRIPTION
**Pull request by Pikachu920**
### Description
Adds a parse structure to allow testing structures for errors. 

---
**Target Minecraft Versions:** any 
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

**Comment by Pikachu920** [comment]
rather than allowing just a single section, i was thinking it might be better to parse anything in the `code` section a separate script (which can contain multiple structures). that way you can test for errors that depend on multiple structures are thrown (e.g. defining two commands of the same name).

File: src/main/java/ch/njol/skript/test/runner/StructParse.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import org.skriptlang.skript.lang.entry.EntryContainer;
+import org.skriptlang.skript.lang.entry.EntryValidator;
+import org.skriptlang.skript.lang.entry.util.ExpressionEntryData;
+import org.skriptlang.skript.lang.structure.Structure;
+
+import static ch.njol.skript.lang.SkriptParser.ParseResult;
+
+
+@Name("Parse Structure")
+@Description("Parses the code inside this structure as a structure and use 'parse logs' to grab any logs from it.")
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/test/runner/StructParse.java:49]: a little more explanation on how to use would be nice. What should I put in results? Do I need a code entry? etc,

**Review by sovdeeth — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]
Very useful feature :clap:

File: src/test/skript/tests/syntaxes/structures/StructParse.sk [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  results: {StructParse::parse::*}
+  code:
+    parse:
+      results: "oops"
+      code:
+        on script load:
+          broadcast "hi"
+
+test "StructParse":
+  assert {StructParse::parse::*} contains """oops"" cannot be set to strings" with "StructParse error not found"
```
Comment (Moderocky) [inline src/test/skript/tests/syntaxes/structures/StructParse.sk:11]: It sounds cool but I really honestly cannot tell what on earth is going on in your test, I think some more explanation or documentation for the feature might help (even some javadoc would be pretty cool).

Comment (AyhamAl-Ali) [reply src/test/skript/tests/syntaxes/structures/StructParse.sk:11]: It's indeed cool! He is using a nested StructParse with invalid store object "oops" that will throw an error and catching it for testing

**Review by Moderocky — APPROVED** [review]

**Pull request merged**